Dockerfile.aro-multistage

```diff
@@ -4,7 +4,7 @@
 # Currently the docker version on our RHEL7 VMSS uses a version which
 # does not support multi-stage builds.  This is a temporary stop-gap
 # until we get podman working without issue
-FROM registry.access.redhat.com/ubi7/go-toolset:1.16.12 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
```

Comment:
@bennerv there are none returned from `podman image search --list-tags registry.access.redhat.com/ubi{7,8}/go-toolset` so I went with the CI version that's used for testing OpenShift 4.10.

I actually suggest we change to building from the latest MSFT Go FIPS source so that going forward we aren't dependant on other parties for our base image. @cadenmarchese I have recently done this to get 1.17 support for our CI Agent Pool, but it sounds like you are already working on solving this with CDPx builds?

Reply:
ubi8 now has 1.17 released: https://catalog.redhat.com/software/containers/ubi8/go-toolset/5ce8713aac3db925c03774d1